clients/roscpp/include/ros/timer_manager.h

```diff
@@ -428,11 +428,36 @@ void TimerManager<T, D, E>::setPeriod(int32_t handle, const D& period)
   {
     return;
   }
+  
+  if(reset)
+  {
+    info->next_expected = T::now() + period;
+  }
+  
+  // If called in a callback, last_real has not been updated yet so (now - last_real) > period
+  // In this case, let next_expected be updated only in updateNext
```

Comment:
Is that actual tested behavior or only assuming the behavior? I thought `updateNext` (https://github.com/ros/ros_comm/blob/indigo-devel/clients/roscpp/include/ros/timer_manager.h#L403-L410) would move the `next_expected` forward in case the callback set it to `now`.